src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java

```diff
@@ -236,12 +151,58 @@ public CoverageViewReader(GridCoverage2DReader delegate, CoverageView coverageVi
     @Override
     public GridCoverage2D read(GeneralParameterValue[] parameters) throws IllegalArgumentException,
             IOException {
+        
+        // did we get a request within the bounds of the coverage?
+        Optional<GeneralParameterValue> ggParameter = Optional.empty();
+        if(parameters != null) {
+            ggParameter = Arrays.stream(parameters).filter(
+                parameter -> parameter.getDescriptor().getName().equals(AbstractGridFormat.READ_GRIDGEOMETRY2D.getName())).findFirst();
+        }
+        GridGeometry2D requestedGridGeometry = null;
+        if(ggParameter.isPresent()) {
+            ParameterValue value = (ParameterValue) ggParameter.get();
+            requestedGridGeometry = (GridGeometry2D) value.getValue();
+            ReferencedEnvelope requestedEnvelope = ReferencedEnvelope.reference(requestedGridGeometry.getEnvelope());
+            ReferencedEnvelope dataEnvelope = ReferencedEnvelope.reference(handler.getOriginalEnvelope());
+            if(CRS.equalsIgnoreMetadata(requestedEnvelope, dataEnvelope)) {
+                if(!requestedEnvelope.intersects((BoundingBox) dataEnvelope)) {
+                    return null;
+                }
+            } else {
+                ReferencedEnvelope re84;
+                try {
+                    re84 = requestedEnvelope.transform(DefaultGeographicCRS.WGS84, true);
+                    ReferencedEnvelope de84 = dataEnvelope.transform(DefaultGeographicCRS.WGS84, true);
+                    if(!re84.intersects((BoundingBox) de84)) {
+                        return null;
+                    }
+                } catch (TransformException | FactoryException e) {
+                    LOGGER.log(Level.FINE, "Cannot determine if the requested BBOX intersects the "
+                            + "data one, continuing", e);
+                    
+                }
+            }
+            
+            // expand the read area if we are in the heterogeneous case... it's a resampling
+            // one similar to reprojection, prone to off-by-one issues at the borders
+            if(!handler.isHomogeneousCoverages()) {
+                GridEnvelope2D range = requestedGridGeometry.getGridRange2D();
+                GridEnvelope2D expandedRange = new GridEnvelope2D(
+                        (int) range.getMinX() - HETEROGENEOUS_RASTER_GUTTER,
+                        (int) range.getMinY() - HETEROGENEOUS_RASTER_GUTTER,
+                        (int) range.getWidth() + HETEROGENEOUS_RASTER_GUTTER * 2,
+                        (int) range.getHeight() + HETEROGENEOUS_RASTER_GUTTER * 2);
+                GridGeometry2D expandedGG = new GridGeometry2D(expandedRange,
+                        requestedGridGeometry.getGridToCRS(),
+                        requestedGridGeometry.getCoordinateReferenceSystem());
+                value.setValue(expandedGG);
+            }
+        }
```

Comment:
small note:
there are 4 if(condition) here not respecting the template 
if (condition)

Reply:
Heh,there were a ton in this file, fixed them all.